packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -965,3 +996,305 @@ export const setEditing = (payload: string | undefined): MetadataAction => ({
+                ? cloneObject(METADATA.DEFAULT_ACCOUNT_METADATA)
+                : cloneObject(METADATA.DEFAULT_WALLET_METADATA);
+
+        const nextData =
+            prevData && 'data' in prevData
+                ? { ...prevData.data, migratedFrom: prevKeys.fileName }
+                : { ...defaultEntityData, ...dummy };
+
+        const providerInstance = dispatch(
+            getProviderInstance({ clientId: getState().metadata.providers[0]!.clientId }),
+        );
+
+        if (!providerInstance) {
+            // provider should always be set here
+            return { success: false, error: 'provider not connected' };
+        }
+
+        dispatch(
+            setMetadata({
+                ...nextKeys,
+                data: nextData,
+                provider: getState().metadata.providers[0]!,
+            }),
+        );
+        const saveResult = await dispatch(
+            encryptAndSaveMetadata({
+                ...nextKeys,
+                data: nextData,
+                provider: getState().metadata.providers[0]!,
+            }),
+        );
+
+        if (!saveResult.success) {
+            return saveResult;
+        }
+
+        // we were only creating dummy
+        if (!fetchData) {
+            return { success: true };
+        }
+
+        // rename only if next version was saved correctly
+        return providerInstance.renameFile(
+            prevKeys.fileName,
+            prevKeys.fileName.replace('.mtdt', '_v1.mtdt'),
+        );
+    };
+
+/**
+ * Check whether encryption version migration is needed and if yes execute it
+ */
+const handleEncryptionVersionMigration =
+    (deviceState: string) =>
+    async (
+        dispatch: Dispatch,
+        getState: GetState,
+    ): Promise<{ success: boolean; error?: string }> => {
+        // 1. select lower encryption version
+        const prevEncryptionVersion = (METADATA.ENCRYPTION_VERSION -
+            1) as MetadataEncryptionVersion;
+        if (prevEncryptionVersion < 1) {
+            return {
+                success: false,
+                error: `metadata migration: can not migrate to version ${prevEncryptionVersion}`,
+            };
+        }
+
+        let device = selectDeviceByState(getState(), deviceState);
+
+        if (!device) {
+            // should never happen
+            return { success: false, error: 'metadata migration: device not found' };
+        }
+
+        // 3. fetch list of all files saved withing currently selected provider for labeling. based on file suffix we are
+        //    able to determine which files are associated with which encryption version
+        const [currentEncryptionFiles, oldEncryptionFiles, renamedOldEncryptionFiles] =
+            await dispatch(getMetadataFiles());
+        console.log('currentEncryptionFiles', currentEncryptionFiles);
+        console.log('oldEncryptionFiles', oldEncryptionFiles);
+        console.log('renamedOldEncryptionFiles', renamedOldEncryptionFiles);
```

Comment:
lefovers (but of course useful for now... just don't forget to remove them before merge)